apps/web/components/apps/installation/ConfigureStepCard.tsx

```diff
@@ -116,82 +117,139 @@ const EventTypeAppSettingsForm = forwardRef<HTMLButtonElement, EventTypeAppSetti
+  const { control } = useFormContext<TEventTypesForm>();
   const { fields, update } = useFieldArray({
     control,
-    name: "eventTypes",
+    name: `eventTypeGroups.${groupIndex}.eventTypes`,
+    keyName: "fieldId",
+  });
+  return (
+    <div className="ml-2 flex flex-col space-y-6">
+      {fields.map(
+        (field, index) =>
+          field.selected && (
+            <EventTypeAppSettingsForm
+              key={field.fieldId}
+              eventType={field}
+              loading={props.loading}
+              handleDelete={() => {
+                const eventTypeDb = eventTypeGroups[groupIndex].eventTypes?.find(
+                  (eventType) => eventType.id == field.id
+                );
+                update(index, {
+                  ...field,
+                  selected: false,
+                  metadata: eventTypeDb?.metadata,
+                  bookingFields: eventTypeDb?.bookingFields,
+                  ...(eventTypeDb?.locations && { locations: eventTypeDb.locations }),
+                });
+
+                setUpdatedEventTypesStatus((prev) => {
+                  const res = [...prev];
+                  res[groupIndex] = res[groupIndex].filter((item) => !(item.id === field.id));
+                  if (!res.some((item) => item.length > 0)) {
+                    props.setConfigureStep(false);
+                  }
+                  return res;
+                });
+              }}
+              onSubmit={(data) => {
+                update(index, { ...field, ...data });
+                setUpdatedEventTypesStatus((prev) => {
+                  const res = [...prev];
+                  res[groupIndex] = res[groupIndex].map((item) =>
+                    item.id === field.id ? { ...item, updated: true } : item
+                  );
+                  return res;
+                });
+              }}
+              ref={submitRefs[index]}
+              {...props}
+            />
+          )
+      )}
+    </div>
+  );
+};
+
+export const ConfigureStepCard: FC<ConfigureStepCardProps> = (props) => {
+  const { loading, formPortalRef, handleSetUpLater } = props;
+  const { t } = useLocale();
+  const { control, watch } = useFormContext<TEventTypesForm>();
+  const { fields } = useFieldArray({
+    control,
+    name: "eventTypeGroups",
     keyName: "fieldId",
   });
+  const eventTypeGroups = watch("eventTypeGroups");
+  const submitRefs = useRef<Array<Array<React.RefObject<HTMLButtonElement>>>>([]);
 
-  const submitRefs = useRef<Array<React.RefObject<HTMLButtonElement>>>([]);
-  submitRefs.current = fields.map(
-    (_ref, index) => (submitRefs.current[index] = React.createRef<HTMLButtonElement>())
+  submitRefs.current = eventTypeGroups.reduce(
+    (arr: Array<Array<React.RefObject<HTMLButtonElement>>>, field) => {
+      const res = field.eventTypes
+        .filter((eventType) => eventType.selected)
+        .map((_ref) => React.createRef<HTMLButtonElement>());
+      return [...arr, res];
+    },
+    []
```

Comment:
don't you a think a lot of code here is complex? There are too many forms and when you click on save it runs a side effect and also submits many hidden forms and parent form.

Reply:
is there any way to implement this?

Reply:
Yeah, I agree it turned out much worse than I thought, 
In my defense, I had to do all this because the react-hook-form doesn't support a form within another form very well.

Reply:
> is there any way to implement this?

I could only think of one, but I’m open to suggestions if any 🙏 .